src/main/java/servlets/Calculator.java

```diff
@@ -0,0 +1,147 @@
+package servlets;
+
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.util.Stack;
+
+
+@WebServlet(
+        name = "Calculator",
+        urlPatterns = {"/calc"}
+)
+public class Calculator extends HttpServlet {
+    private Stack<String>  stack = new Stack<>();
+    private Stack<String>  nstack = new Stack<>();
+    private int prev = 0;
+
+    private void stackaction() {
+        stack.push(String.valueOf(nstack.peek()));
+        nstack.pop();
+    }
+
+    private boolean equals(String s) {
+        return nstack.peek().equals(s);
+    }
+
+    private void equationtostack(char[] modequation) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'equationtostack(char)' has an NPath complexity of 588](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4500113)